essentials-framework/Essentials Core/PepperDashEssentialsBase/Factory/DeviceFactory.cs

```diff
@@ -83,33 +85,78 @@ public static void AddFactoryForType(string typeName, string description, CType
 
             var wrapper = new DeviceFactoryWrapper() { CType = cType, Description = description, FactoryMethod = method };
             DeviceFactory.FactoryMethods.Add(typeName, wrapper);
-        }
-
-		/// <summary>
-		/// The factory method for Core "things". Also iterates the Factory methods that have
-		/// been loaded from plugins
-		/// </summary>
-		/// <param name="dc"></param>
-		/// <returns></returns>
-        public static IKeyed GetDevice(DeviceConfig dc)
-        {
-            var key = dc.Key;
-            var name = dc.Name;
-            var type = dc.Type;
-            var properties = dc.Properties;	
-
-            var typeName = dc.Type.ToLower();
-
-            // Check for types that have been added by plugin dlls. 
-            if (FactoryMethods.ContainsKey(typeName))
-            {
-                Debug.Console(0, Debug.ErrorLogLevel.Notice, "Loading '{0}' from Essentials Core", dc.Type);
-                return FactoryMethods[typeName].FactoryMethod(dc);
-            }
-
-            return null;
-        }
-
+        }
+
+        private static void CheckForSecrets(IEnumerable<JProperty> obj)
+        {
+            foreach (var prop in obj.Where(prop => prop.Value as JObject != null))
+            {
+                if (prop.Name.ToLower() == "secret")
+                {
+                    var secret = GetSecret(JsonConvert.DeserializeObject<SecretsPropertiesConfig>(prop.Children().First().ToString()));

```

Comment:
you should be able to use `prop.children().First.ToObject<SecretsPropertiesConfig>` here instead of serializing to a string and deserializing again.